### DISCUSSION TITLE
FEATURE REQUEST: category for launchers

### DISCUSSION THREAD

**Discussion by spamegg1** [linuxmint/discussions#475] [Ideas]
On Cinnamon, when we right-click on the Desktop and choose "Create a new launcher here..." it offers us an option to also add the launcher to the Menu. It says that it will be first added to the "Other category."

This is where things start to go wrong. A lot of newcomers from Windows try to manually "fix" this via the Menu Editor. (I've been helping many forum users with this.)

But the Menu Editor is very buggy. Even when it's not, it does not work like people expect at all.

So... why not give us the option to actually choose the Menu category when we are creating the Launcher? This side-steps the Menu editor and its various problems.

Just add a "Category" drop-down menu here, which should add the relevant line of text to the `.desktop` file:
![Screenshot from 2024-05-30 21-46-15](https://github.com/linuxmint/discussions/assets/4255997/00320546-a1d2-4aad-9067-563bc8f8f7f1)

**Comment by billyswong** [comment]
A .desktop launcher file is specified in a way that the "Categories" of a launcher can be arbitrary and also multi-select. So a proper GUI for "Categories" management shall
1. preserve category names already specified in the launcher file that Cinnamon doesn't understand
2. let one add/remove categories that Cinnamon understand with a list of checkboxes.

The existing cinnamon-menu-editor UI seems to be forked from MATE Mozo. That interface is problematic as it tries to hide the actual architecture of Linux GUI menu and pretend / abstract it as something else. But then, the pretension / abstraction fails miserably when one really tries to do any manipulation.

An extra quirk: The "Categories" parameter supports sub-menu for classical menu design. The Launcher Properties dialog need to handle them correctly too.

**Reply by spamegg1** [reply]
Very interesting.

I think it would be sufficient to just provide the basic categories that come with the Cinnamon menu when Mint is first installed. It doesn't have to be super comprehensive.

My goal is to just make beginners avoid having to manually try to move it from "Other" to something else, which is where the problems start.

**Comment by f1gwr** [comment]
Agree with this request: numerous bugs in Menu Editor
